src/analytics/projectSettings/useProjectSettingsAnalytics.ts

```diff
@@ -0,0 +1,36 @@
+import { useParams } from "react-router-dom";
+import {
+  addPageAction,
+  Properties,
+  Analytics as A,
+} from "analytics/addPageAction";
+import { useGetUserQuery } from "analytics/useGetUserQuery";
+
+type Action =
+  | { name: "Save project"; section: string }
+  | { name: "Save repo"; section: string }
+  | { name: "Default section to repo"; section: string }
+  | { name: "Attach project to repo"; repoOwner: string; repoName: string }
+  | { name: "Detach project from repo"; repoOwner: string; repoName: string }
+  | { name: "Move project to new repo"; repoOwner: string; repoName: string };
+
+interface P extends Properties {
+  identifier: string;
+}
+
+export interface Analytics extends A<Action> {}
+
+export const useProjectSettingsAnalytics = (): Analytics => {
+  const userId = useGetUserQuery();
```

Comment:
Realized we did this recently. We actually set userid globally for new relic. I opened https://jira.mongodb.org/browse/EVG-17035 to get rid of it everywhere else.